### PR TITLE
[internal/rabbitmq] move connection and retry logic into separate pkg

### PR DESCRIPTION
**Pull request by frzifus**
**Description:** <Describe what has changed.>

This PR moves the retry logic from amqp publisher to amqp connection. Connection, client and other utility structures have been moved from `exporter/rabbitmqexporter/internal/publisher` to `internal/rabbitmq`.

**Link to tracking Issue:** 
- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34242

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

----

cc @swar8080 @atoulme

### CODE REVIEW

**Review by swar8080 — COMMENTED** [review]
looks good! just added a couple minor comments

File: exporter/rabbitmqexporter/internal/publisher/publisher_test.go [hunk 1]
```diff
@@ -307,15 +316,20 @@ type mockConnection struct {
 	mock.Mock
 }
 
+func (m *mockConnection) ReconnectIfUnhealthy() error {
```
[the remark this answers is not in the document]
Comment (frzifus) [reply exporter/rabbitmqexporter/internal/publisher/publisher_test.go:319]: Agree. I added a test in the `internal/rabbitmq` pkg.

File: exporter/rabbitmqexporter/internal/publisher/publisher.go [hunk 2]
```diff
@@ -5,26 +5,20 @@ package publisher // import "github.com/open-telemetry/opentelemetry-collector-c
...
-	"crypto/tls"
 	"errors"
 	"fmt"
-	"sync"
 	"time"
 
 	amqp "github.com/rabbitmq/amqp091-go"
 	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/rabbitmq"
```
[the remark this answers is not in the document]
Comment (frzifus) [reply exporter/rabbitmqexporter/internal/publisher/publisher.go]: Changed it :+1:

**Review by swar8080 — APPROVED** [review]

**Review by swar8080 — APPROVED** [review]

**Review by iblancasa — APPROVED** [review]

**Review by mwear — APPROVED** [review]
LGTM. @atoulme, do mind taking a look when you get a chance?

**Review by atoulme — APPROVED** [review]
LGTM!

**Comment by frzifus** [comment]
FTR: just resolved merge conflicts in the `go.mod` files.

File: internal/rabbitmq/go.mod [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/internal/rabbitmq
+
+go 1.21.0
```
Comment (codeboten) [inline internal/rabbitmq/go.mod:3]:
```suggestion
go 1.22.0
```

**Review by codeboten — APPROVED** [review]

**Pull request merged**